Lib/pydoc.py

```diff
@@ -1617,13 +1618,13 @@ def pipepager(text, cmd):
 def tempfilepager(text, cmd):
     """Page through text by invoking a program on a temporary file."""
     import tempfile
-    filename = tempfile.mktemp()
-    with open(filename, 'w', errors='backslashreplace') as file:
-        file.write(text)
     try:
-        os.system(cmd + ' "' + filename + '"')
+        with tempfile.NamedTemporaryFile('w', delete=False, errors='backslashreplace') as file:
+            file.write(text)
+            file.close()
+            subprocess.run([cmd, file.name])
```

Comment:
Replacing `os.system()` with `subprocess.run()` here seems out of scope of this pull request per the bpo description.

Reply:
Fair. I just felt it might be worth doing while we were working on it (the existing method is awful). I will revert this.